SparseGrids/tasgridUnitTests.cpp

```diff
@@ -786,4 +800,69 @@ std::vector<std::function<void(void)>> GridUnitTester::getRuntimeErrorCalls() co
     };
 }
 
+bool GridUnitTester::testLAPACKInterface() {
+
+    bool all_matched = true;
+    #ifdef Tasmanian_ENABLE_BLAS
+    // Initialize.
+    const int N = 100;
+    double* exact_eigs = new double[N];
+    double* D = new double[N];
+    double* E = new double[N-1];
+    double* W = new double[N];
+    double* Z = new double[N*N];
+    double* WORK1 = new double[2*N-2];
+    double* WORK2 = new double[4*N];
+    int* IBLOCK1 = new int[N];
+    int* ISPLIT1 = new int[N];
+    int* IWORK1 = new int[3*N];
+    double a = TasGrid::Maths::pi / 2;
+    double b = TasGrid::Maths::pi / 4;
+    ISPLIT1[0] = N;
+    IBLOCK1[0] = N;
+    for (int i=0; i<N; i++) {
+        exact_eigs[i] = a + 2.0 * b * std::cos((i+1) * TasGrid::Maths::pi / (N+1));
+    }
+    std::sort(exact_eigs, exact_eigs + N);
+
+    // Test LAPACK's dsterf function.
+    std::fill_n(D, N, a);
+    std::fill_n(E, N-1, b);
+    TasBLAS::sterf(N, D, E);
+    std::sort(D, D + N);
+    all_matched = all_matched && doesMatch(N, D, exact_eigs);
+
+    // Test LAPACK's dsteqr function.
+    std::fill_n(D, N, a);
+    std::fill_n(E, N-1, b);
```

Comment:
Are `D` and `E` destroyed in the previous call. Why do you need to initialize again?

Reply:
According to the LAPACK documentation (http://www.netlib.org/lapack/explore-html/d2/d24/group__aux_o_t_h_e_rcomputational_gaf0616552c11358ae8298d0ac18ac023c.html), D is both an input (diagonal entries) and output (computed eigenvalues) while the contents of E are destroyed in the previous call.

Reply:
Note that the person who starts the conversation should be the one to hit the "resolve" button. Once the conversation is resolved, github hides it and I cannot see your answer unless I find it and click to unhide.

Otherwise, you're right here and you have to refill the vector, but the first fill can still be done in the initialization stage. Also, make sure that the C++ wrappers properly reflect the const or non-const aspect of a variable.

Reply:
Gotcha. I'll keep all future conversations open.